warehouse/http.py

```diff
@@ -0,0 +1,34 @@
+import threading
+import requests
+
+
+_local = threading.local()
```

Comment:
Yeah, I was on the fence about this one. The thing that I like about the global is that it's a little more resilient to external changes then. i.e. If someone was to set `reify` to `False` for some reason, then we'd start getting a new session per call. Minor as it really /shouldn't/ happen, but that was the reasoning. In any event, I've updated it.